service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RespondentControllerAboutToSubmitTest.java

```diff
@@ -0,0 +1,211 @@
+                .build())
+            .build();
+        assertThat(responseData.getNoticeOfChangeAnswersData()).isEqualTo(expectedAnswers);
+    }
+
+    @Test
+    void shouldRemoveLegalCounselFromRespondentWhenRepresentativeIsRemoved() {
+        List<Element<LegalCounsellor>> counsel = List.of(element(buildLegalCounsellor("1", true)));
+        Respondent respondent1 = respondent(dateNow()).toBuilder()
+            .solicitor(TEST_RESPONDENT_SOLICITOR)
+            .legalCounsellors(counsel)
+            .build();
+        Respondent respondent2 = respondent(dateNow()).toBuilder()
+            .solicitor(TEST_RESPONDENT_SOLICITOR)
+            .legalCounsellors(counsel)
+            .build();
+
+        CaseData caseDataBefore = CaseData.builder()
+            .applicants(TEST_APPLICANTS)
+            .respondents1(wrapElements(respondent1, respondent2))
+            .build();
+        CaseData caseDataAfter = CaseData.builder()
+            .applicants(TEST_APPLICANTS)
+            .respondents1(wrapElements(respondent1.toBuilder().solicitor(null).build(), respondent2))
+            .build();
+
+        CallbackRequest callbackRequest = toCallBackRequest(caseDataAfter, caseDataBefore);
+        CaseData returnedCaseData = extractCaseData(postAboutToSubmitEvent(callbackRequest));
+
+        assertThat(returnedCaseData.getAllRespondents().get(0).getValue().getLegalCounsellors()).isEmpty();
+        assertThat(returnedCaseData.getAllRespondents().get(1).getValue().getLegalCounsellors()).isEqualTo(counsel);
+    }
+
+    @Test
+    void shouldPersistRepresentativeAssociation() {
+        List<Element<UUID>> association = List.of(element(UUID.randomUUID()));
+        Element<Respondent> oldRespondent = element(respondent(dateNow()));
+        oldRespondent.getValue().setRepresentedBy(association);
+
+        CaseData caseData = CaseData.builder()
+            .respondents1(List.of(
+                element(oldRespondent.getId(), respondent(dateNow())),
+                element(respondent(dateNow()))
+            ))
+            .state(OPEN)
+            .build();
+
+        CaseData caseDataBefore = CaseData.builder()
+            .respondents1(List.of(oldRespondent))
+            .build();
+
+        CallbackRequest callbackRequest = toCallBackRequest(caseData, caseDataBefore);
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(callbackRequest);
+
+        CaseData responseData = extractCaseData(response);
+
+        Respondent firstRespondent = responseData.getRespondents1().get(0).getValue();
+        Respondent secondRespondent = responseData.getRespondents1().get(1).getValue();
+
+        assertThat(firstRespondent.getRepresentedBy()).isEqualTo(association);
+        assertThat(secondRespondent.getRepresentedBy()).isNullOrEmpty();
+    }
+
+    @Test
+    void aboutToSubmitShouldAddConfidentialRespondentsToCaseDataWhenConfidentialRespondentsExist() {
```

Comment:
Could remove the "aboutToSubmit" prefix from the test name to make it more consistent with the other test names?